src/index.ts

```diff
@@ -37,10 +41,157 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
+  return twiddleData;
+}
+
+/**
+ * Normalize the generated twiddle to a ember-like structure.
+ * @param targetDir
+ */
+function normalizeTwiddle(targetDir: string) {
+  const appDir = `${targetDir}/app`;
+  generateDir(appDir);
+  const files = fs.readdirSync(targetDir);
+  for (let i in files) {
+    const file = files[i];
+    const dirPath = file.split('.');
+    const fileName = dirPath.splice(-2, 2);
+    const newPath = `${appDir}/${dirPath.join('/')}`;
+    const filePath = `${newPath}/${fileName.join('.')}`;
+    // If the file parts length is more than 2 then there is a directory structure expected.
+    if (file.split('.').length > 2) {
+      generateDir(`${appDir}/${dirPath.join('/')}`);
+    }
+    if (fileName.length === 2 && fs.existsSync(newPath)) {
+      try {
+        if (fs.existsSync(filePath)) {
+          continue;
+        } else {
+          fs.renameSync(`${targetDir}/${file}`, filePath);
+        }
+      } catch (err) {
+        throw err;
+      }
+    }
+  }
+}
+
+/**
+ * Augment the already generated ember-like twiddle project app into a real ember app
+ * by augmenting the real Ember app's structure on top of the twiddle app.
+ * @param twiddleData
+ * @param projectName
+ */
+async function augmentEmberApp(twiddleData: any, projectName: string) {
+  const twiddleItem = JSON.parse(JSON.stringify(twiddleData));
+  const emberVersion = twiddleItem.dependencies.ember.split('.');
+  emberVersion.pop();
+  const emberAppDir = await unzipAndExtractFromUrl(
+    `https://${GIT_DOMAIN}/ember-cli/ember-new-output`,
+    `archive/v${emberVersion.join('.')}.0.zip`,
+    `./${projectName}`
+  );
+  if (emberAppDir) {
+    const packageData = getJsonFileData(emberAppDir, 'package');
+    const packageDevDep = packageData.devDependencies;
+    const twiddleAddons = twiddleData.addons;
+    let isDevDependencyInTwiddle = false;
+    Object.keys(twiddleAddons).forEach(item => {
+      if (!packageDevDep.hasOwnProperty(item)) {
+        isDevDependencyInTwiddle = true;
+        packageDevDep[item] = twiddleAddons[item];
+      }
+    });
+    // If there are dev dependencies present in twiddle, add them to the package json of the newly generated ember app.
+    if (isDevDependencyInTwiddle) {
+      fs.writeFileSync(`${emberAppDir}/package.json`, JSON.stringify(packageData));
+    }
+    // tslint:disable-next-line:no-console
+    console.log('Project converted succesfully!');
+  }
+}
+
+export default async function twiddleEject(twiddleUrl: string, projectName: string, _options: {} = {}) {
   const gistId = gistIdFromTwiddleUrl(twiddleUrl);
   // tslint:disable-next-line:no-console
   console.log('gistId', gistId);
   // tslint:disable-next-line:no-console
   console.log('project name', projectName);
+
+  // Unzip and extract files from the gist url.
+  const targetDir = await unzipAndExtractFromUrl(
+    `https://${GIST_DOMAIN}/${gistId}`,
+    'archive/HEAD.zip',
```

Comment:
```suggestion
    path.join(archive, 'HEAD.zip'),
```

Reply:
I purposely didn't do it for this one since this part is used for generating the url for fetching the zip file and not actually `path` for a folder structure